base/common/common_utils/log_monitor.py

```diff
@@ -0,0 +1,168 @@
+import json
+import re
+from datetime import datetime
+from typing import Match, Optional, List
+
+from picsellia.types.enums import JobRunStatus, ExperimentStatus
+
+from .log_tailer import LogTailer
+from .picsellia_utils import (
+    get_picsellia_job,
+    get_picsellia_client,
+    get_picsellia_experiment,
+)
+
+
+def starts_buffer(line: str) -> Match[str] | None:
```

Comment:
I would put the variables inside constants, and rename the functions to better match their purpose, something like:

```python
# Constants for better clarity and avoiding magic strings
BUFFER_START_PATTERN = re.compile(r"--[0-9]--")
BUFFER_END_PATTERN = re.compile(r"---[0-9]---")

def does_buffer_start(line: str) -> Match[str] | None:
    """Check if line starts a buffer."""
    return BUFFER_START_PATTERN.match(line[:6])


def does_buffer_end(line: str) -> Match[str] | None:
    """Check if line ends a buffer."""
    return BUFFER_END_PATTERN.match(line[:8])
```